### ISSUE TITLE
The question about the loss fuction

### ISSUE THREAD

**Issue by mingsiliu557** [ManiadisG/DivClust#2]
Dear author,
 I have a problem about the loss function. As shown in the paper, we use clusters' probability distributions to calculate the loss, however, use the probability distributions of samples to calculate DR, in order to upgrade the threshold d. Why not use the same distributions to calculate the loss and DR?

**Comment by ManiadisG** [comment]
Hi!

For the update of DR we don't actually use probability distributions. We convert the probability distributions of samples-to-clusters into hard assignments, and use them to measure the NMI between clusterings.

The reason is that, while our loss is applied in the distribution space, if we calculated DR there the confidence of sample assignments would be a factor in whichever similarity measure we chose and that would cause problems. For example: if we used cosine similarity as in the loss, the degree of similarity between clusterings would change a lot depending on the sharpness of cluster distributions (even without the "hard" assignment being any different). Therefore, any threshold a user would choose would have a different impact depending on the stage of training and assignment confidence. We found that using hard assignments to measure DR worked better, was much more stable, and also that it is much more intuitive for users

I'm not sure if that answers your question fully, please let me know if I've missed something.

**Comment by mingsiliu557** [comment]
Thanks, I fully get it!
